backend/src/tools/api.py

```
@@ -0,0 +1,515 @@
+                },
+            }
+        }
+
+def get_api_spec(
+    title: str,
+    version: str,
+    base_url: str,
+    endpoints: List[str],
+    common_query_parameters: List[Dict[str, Any]],
+    description: str,
+    paths: Dict[str, Any] = {},
```

Ruff (0.11.9)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)